src/table.rs

```diff
@@ -236,6 +238,77 @@ impl Table {
         self.rows.len()
     }
 
+    /// Sorts the rows of the table using a custom comparator function.
+    ///
+    /// The sorting is done in place and modifies the original table.
+    ///
+    /// ## Example
+    ///
+    /// The following example demonstrates how to sort the rows of a table
+    /// in descending order based on the content of the first column:
+    ///
+    /// ```
+    /// // Create a new table with a header and several rows
+    /// let mut table = Table::new();
+    /// table
+    ///     .set_header(vec![Cell::new("Names")])
+    ///     .add_row(vec![Cell::new("Alexander the Great")])
+    ///     .add_row(vec![Cell::new("Shahd Aldahar")])
+    ///     .add_row(vec![Cell::new("Neil Armstrong")])
+    ///     .add_row(vec![Cell::new("Hassan Ibn Al-haitham")])
+    ///     .add_row(vec![Cell::new("Al-khwarizmi")]);
+    ///
+    /// // Sort the rows in descending order based on the first column
+    /// table.sort_rows(|a, b| b.cells()[0].cmp(&a.cells()[0]));
+    ///
+    /// // Print the table after sorting
+    /// println!("Sort in descending order:");
+    /// println!("{table}");
+    /// ```
+    ///
+    /// **Output:**
+    /// ```plaintext
+    /// Sort in descending order:
+    /// +-----------------------+
+    /// | Names                 |
+    /// +=======================+
+    /// | Shahd Aldahar         |
+    /// |-----------------------|
+    /// | Neil Armstrong        |
+    /// |-----------------------|
+    /// | Hassan Ibn Al-haitham |
+    /// |-----------------------|
+    /// | Alexander the Great   |
+    /// |-----------------------|
+    /// | Al-khwarizmi          |
+    /// +-----------------------+
+    /// ```
+    ///
```

Comment:
Nice docs, but let's try to keep it short and simple :)
You already described how the result will look like above, so you can drop the output :)